src/core/server/metrics/metrics_service_analytics/ops_metrics_event_schema.ts

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { RootSchema } from '@kbn/analytics-client';
+import { OpsMetricsEvent } from '.';
+import type { OpsProcessMetrics } from '../collectors';
+
+/**
+ * Schema expected for a single process contained in ops metrics events 'processes' array.
+ * @internal
+ */
+const processSchema: RootSchema<OpsProcessMetrics> = {
+  pid: {
+    type: 'long',
+    _meta: { description: 'pid of the kibana process' },
+  },
+  memory: {
+    properties: {
+      heap: {
+        properties: {
+          total_in_bytes: {
+            type: 'byte',
+            _meta: { description: 'total heap available' },
+          },
+          used_in_bytes: {
+            type: 'byte',
+            _meta: { description: 'used heap' },
+          },
+          size_limit: {
+            type: 'byte',
+            _meta: { description: 'v8 heap size limit' },
+          },
+        },
+      },
+      resident_set_size_in_bytes: {
+        type: 'byte',
+        _meta: { description: 'node rss' },
+      },
+    },
+    _meta: {
+      description: 'heap memory usage',
+      optional: false,
+    },
+  },
+  event_loop_delay: {
+    type: 'double',
+    _meta: { description: 'mean event loop delay since last collection' },
+  },
+  event_loop_delay_histogram: {
+    properties: {
+      fromTimestamp: {
+        type: 'date',
+        _meta: {
+          description:
+            'The first timestamp the interval timer kicked in for collecting data points.',
+        },
+      },
+      lastUpdatedAt: {
+        type: 'date',
+        _meta: {
+          description: 'Last timestamp the interval timer kicked in for collecting data points.',
+        },
+      },
+      min: {
+        type: 'double',
+        _meta: { description: 'The minimum recorded event loop delay.' },
```

Comment:
Specifying the unit here makes a lot of sense, especially because we're [converting](https://github.com/elastic/kibana/blob/main/src/core/server/metrics/event_loop_delays/event_loop_delays_monitor.ts#L61) nanoseconds to milliseconds in the collector.